pippo-core/src/main/java/ro/fortsoft/pippo/core/Response.java

```diff
@@ -388,14 +388,64 @@ public boolean isCommitted() {
         return httpServletResponse.isCommitted();
     }
 
+    /**
+     * Attempts to negotiate the Response content type based on the Request
+     * accept type. If an agreement can't be reached, the Response content-type
+     * is not set.
+     *
+     * This allows a response chain such as
+     * <code>response.xml().negotiate(request).send(myObject);</code> which will
+     * set the content type as <code>application/xml</code> and optionally
+     * override it after negotation.
+     *
+     * @param request
+     * @return the response
+     */
+    public Response negotiate(Request request) {
```

Comment:
I have an idea to set the content type in constructor using the request or the accept type (String) - add a new parameter in constructor - but I don't know if it's valuable.